### DISCUSSION TITLE
WagtailSearch database in chinese

### DISCUSSION THREAD

**Discussion by becory** [wagtail/wagtail#12417] [Q&A]
Hi, Everyone:

Thank you for your reading.

I was using wagtail 2.16.2 and updgrade to wagtail 3.0.3.
My website is all Traditional Chinese version.
I found the search backend upgrade and the search is not good at all in wagtail admin portal.

I am trying to find some solutions to resolve this case.
I found it may can help me to improve PostgresSQL text search in Chinese.
[fulltext-search-with-wagtail-and-postgresql](https://lerry.me/post/2023/04/fulltext-search-with-wagtail-and-postgresql)

I can't install zhparser because of I use AWS RDS to manage my database. 

1. How can I resolve this question?
2. If I can't resolve this question , how can I replace the search backend just for search title__icontains for all in wagtail admin panel?

Thank you for your help.

Version: wagtail 3.0.3
PostgresSQL: AWS RDS postgres 12

**Comment by becory** [comment]
In the end, I use "fallback" backend.
```
"BACKEND": "wagtail.search.backends.database.fallback"
```

**Comment by Ganesh-403** [comment]
@becory The behavior you're seeing is unfortunately expected.

When Wagtail moved from the older PostgreSQL search backend to the unified database search backend (introduced in Wagtail 2.15 and made the default in Wagtail 3.0), it relies on PostgreSQL's native full-text search configuration. For Chinese (including Traditional Chinese), PostgreSQL's built-in tokenizer is quite limited, so without an extension such as **zhparser**, search quality is generally poor. :contentReference[oaicite:0]{index=0}

Since you're using **AWS RDS**, installing `zhparser` isn't an option because it requires a PostgreSQL extension that's not available on standard RDS instances.

So your options are essentially:

1. **Use the database fallback backend**

   ```python
   WAGTAILSEARCH_BACKENDS = {
       "default": {
           "BACKEND": "wagtail.search.backends.database.fallback",
       }
   }
   ```

   This falls back to simpler database lookups instead of PostgreSQL FTS. It's less sophisticated but can actually work better for languages where PostgreSQL tokenization is weak. :contentReference[oaicite:1]{index=1}

2. **Use an external search backend**

   If search quality in the admin is important, I'd recommend using one of Wagtail's supported external backends (OpenSearch or Elasticsearch), which provide much better support for multilingual analyzers. :contentReference[oaicite:2]{index=2}

Regarding your other question:

> *"How can I replace the search backend just for search `title__icontains` in the Wagtail admin panel?"*

From what I could find, there isn't a supported setting to make **only the admin page chooser/search** use `title__icontains` while leaving the rest of Wagtail's search backend unchanged. The admin search uses Wagtail's configured search backend rather than a simple Django ORM filter, and I couldn't find an extension point to swap that behavior globally.

Given your constraints (Traditional Chinese + AWS RDS), I think using the **database fallback backend**, as you mentioned in your follow-up, is a reasonable solution if you don't want to introduce an external search service.

Out of curiosity, how large is your site (roughly how many pages)? If it's only a few thousand pages, the fallback backend is often perfectly adequate. If it's significantly larger, an OpenSearch/Elasticsearch backend would likely provide a much better search experience for Chinese content.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
